quickjs.c

```diff
@@ -53364,6 +53364,13 @@ static JSValue js_typed_array_get_byteOffset(JSContext *ctx,
     return JS_NewInt32(ctx, ta->offset);
 }
 
+JSValue JS_NewTypedArray(JSContext *ctx, int argc, JSValueConst *argv,
+                         JSTypedArrayEnum array_type)
+{
+    return js_typed_array_constructor(ctx, JS_UNDEFINED, argc, argv,
+                                      JS_CLASS_UINT8C_ARRAY + array_type);
+}
```

Comment:
You should add a range check and either return `JS_UNDEFINED` or throw an exception.  The check could be performed in `js_typed_array_constructor` instead.